From c54281f5ca391711c99b78481105262db66850c6 Mon Sep 17 00:00:00 2001
From: Karthik D A <karthikda@maxlinear.com>
Date: Tue, 15 Jul 2025 18:37:59 +0800
Subject: [PATCH] WLANRTSYS-91747 iwlwav-hostapd-ng: QM SCS TCs fixes

Description: The SCS support was added in r1206CV, and the validation
of the feature is achieved by executing the QM certification
SCS TCs 4.5.1_24G/5G, 4.5.2_24G/5G.
Both the TCs failed, and the PR consists of the fix for the same.

Verification: Executed the QM SCS TCs, and TCs passed successfully.

Signed-off-by: Karthik D A <karthikda@maxlinear.com>
---
 src/ap/mxl_sta_info.h            |   6 +
 src/ap/mxl_wmm.c                 | 203 ++++++++++++++++++-------------
 src/ap/mxl_wmm.h                 |   4 +-
 src/common/ieee802_11_defs.h     |   7 +-
 src/common/mxl_ieee802_11_defs.h |  22 ++--
 5 files changed, 137 insertions(+), 105 deletions(-)

diff --git a/src/ap/mxl_sta_info.h b/src/ap/mxl_sta_info.h
index 2726b0f14..0fdaa2a3a 100644
--- a/src/ap/mxl_sta_info.h
+++ b/src/ap/mxl_sta_info.h
@@ -16,6 +16,12 @@
 #include "utils/common.h"
 
 struct hostapd_data;
+struct mxl_scs_status_list
+{
+	u8 scsId;
+	u16 status;
+}__attribute__ ((packed));
+
 struct mxl_sta_info {
 	u8 multi_ap_profile; /* Multi-AP spec agent Profile (1/2) */
 #ifdef CONFIG_LEGACY_MULTI_AP_MXL
diff --git a/src/ap/mxl_wmm.c b/src/ap/mxl_wmm.c
index 5a1fe521b..201384b81 100644
--- a/src/ap/mxl_wmm.c
+++ b/src/ap/mxl_wmm.c
@@ -174,7 +174,7 @@ void mxl_hostapd_eht_protected(struct hostapd_data *hapd,
 
 #ifdef CONFIG_SCS_MXL
 static void mxl_send_scs_resp(struct hostapd_data *hapd,
-			const struct ieee80211_mgmt *mgmt, int status)
+			const struct ieee80211_mgmt *mgmt, struct mxl_scs_status_list *status_list, u8 scs_count)
 {
 	u8 buf[256];
 	struct ieee80211_mgmt *mgmt_resp = (struct ieee80211_mgmt *) buf;
@@ -197,10 +197,12 @@ static void mxl_send_scs_resp(struct hostapd_data *hapd,
 	mgmt_resp->u.action.category = WLAN_ACTION_ROBUST_AV_STREAMING;
 	mgmt_resp->u.action.u.scs_action_resp.action = ROBUST_AV_SCS_RESP;
 	mgmt_resp->u.action.u.scs_action_resp.dialog_token = mgmt->u.action.u.scs_action_req.dialog_token;
-	mgmt_resp->u.action.u.scs_action_resp.count = 1;
-	mgmt_resp->u.action.u.scs_action_resp.scsId = mgmt->u.action.u.scs_action_req.scsId;
-	mgmt_resp->u.action.u.scs_action_resp.status = status;
-	if (hostapd_drv_send_mlme(hapd, mgmt_resp, IEEE80211_HDRLEN + sizeof(mgmt_resp->u.action.u.scs_action_resp) + sizeof(mgmt_resp->u.action.category), 0, NULL, 0, 0) < 0)
+	mgmt_resp->u.action.u.scs_action_resp.count = scs_count;
+	os_memcpy(mgmt_resp->u.action.u.scs_action_resp.scs_status, status_list, (scs_count * sizeof(struct mxl_scs_status_list)));
+
+	if (hostapd_drv_send_mlme(hapd, mgmt_resp, IEEE80211_HDRLEN + sizeof(mgmt_resp->u.action.u.scs_action_resp) +
+									  sizeof(mgmt_resp->u.action.category) +
+									  (scs_count * sizeof(struct mxl_scs_status_list)), 0, NULL, 0, 0) < 0)
 		wpa_printf(MSG_DEBUG, "ROBUST AV scs_action_send_resp: send failed");
 }
 
@@ -288,7 +290,7 @@ int mxl_scs_parse_tclas_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req
 	const u8 *end;
 	u8 class_type, version;
 	u8 mask = 0, tclas_len = 0;
-	bool has_type4 = false, has_type10 = false;
+	bool has_type10 = false;
 
 	tclas_len = len;
 	scs_add_req->tclasInfo.tclasLen = tclas_len;
@@ -300,82 +302,94 @@ int mxl_scs_parse_tclas_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req
 	while (pos < end) {
 		class_type = *pos++;
 		switch (class_type) { /* Classifier type 4 and 10 are allowed */
-			case SCS_CLASSIFIER_TYPE4:
+			case CLASSIFIER_TYPE4:
 				mask = *pos++;
 
 				/* Process classifiers frames */
-				if (mask & BIT(SCS_CLASS_TYPE4_VERSION))
-					scs_add_req->type4Params.version = *pos++; /* Version */
+				if (mask & BIT(CLASS_TYPE4_VERSION))
+					scs_add_req->type4Params.version = *pos; /* Version */
+				pos ++; /* Length of version */
 
 				if (scs_add_req->type4Params.version != IPV4 && scs_add_req->type4Params.version != IPV6)
 					return WLAN_STATUS_UNSPECIFIED_FAILURE;
 
 				version = scs_add_req->type4Params.version;
 
-				if (mask & BIT(SCS_CLASS_TYPE4_SRC_IP)) { /* source IP address */
+				if (mask & BIT(CLASS_TYPE4_SRC_IP)) { /* source IP address */
 					if (version == IPV4) {
 						memcpy(&(scs_add_req->type4Params.u.v4.srcIp), pos, sizeof(u32));
-						pos += sizeof(u32);
 					} else {
 						memcpy(scs_add_req->type4Params.u.v6.srcIp, pos, 16);
-						pos += 16;
 					}
 				}
 
-				if (mask & BIT(SCS_CLASS_TYPE4_DST_IP)) { /* Destination IP address */
+				if (version == IPV4) /* Length of src ip */
+					  pos += sizeof(u32);
+				else
+					  pos += 16;
+
+				if (mask & BIT(CLASS_TYPE4_DST_IP)) { /* Destination IP address */
 					if (version == IPV4) {
 						memcpy(&(scs_add_req->type4Params.u.v4.dstIp), pos, sizeof(u32));
-						pos += sizeof(u32);
 					} else {
 						memcpy(scs_add_req->type4Params.u.v6.dstIp, pos, 16);
-						pos += 16;
 					}
 				}
 
-				if (mask & BIT(SCS_CLASS_TYPE4_SRC_PORT)) { /* Source port */
+				if (version == IPV4) /* Length of dst ip */
+					  pos += sizeof(u32);
+				else
+					  pos += 16;
+
+				if (mask & BIT(CLASS_TYPE4_SRC_PORT)) { /* Source port */
 					if (version == IPV4) {
 						memcpy(&(scs_add_req->type4Params.u.v4.srcPort), pos, sizeof(u16));
-						pos += sizeof(u16);
+						scs_add_req->type4Params.u.v4.srcPort = ntohs(scs_add_req->type4Params.u.v4.srcPort);
 					} else {
 						memcpy(&(scs_add_req->type4Params.u.v6.srcPort), pos, sizeof(u16));
-						pos += sizeof(u16);
+						scs_add_req->type4Params.u.v6.srcPort = ntohs(scs_add_req->type4Params.u.v6.srcPort);
 					}
 				}
+				pos += sizeof(u16); /* Length of src port */
 
-				if (mask & BIT(SCS_CLASS_TYPE4_DST_PORT)) { /* Dst port */
+				if (mask & BIT(CLASS_TYPE4_DST_PORT)) { /* Dst port */
 					if (version == IPV4) {
 						memcpy(&(scs_add_req->type4Params.u.v4.dstPort), pos, sizeof(u16));
-						pos += sizeof(u16);
+						scs_add_req->type4Params.u.v4.dstPort = ntohs(scs_add_req->type4Params.u.v4.dstPort);
 					} else {
 						memcpy(&(scs_add_req->type4Params.u.v6.dstPort), pos, sizeof(u16));
-						pos += sizeof(u16);
+						scs_add_req->type4Params.u.v6.dstPort = ntohs(scs_add_req->type4Params.u.v6.dstPort);
 					}
 				}
+				pos += sizeof(u16); /* Length of dst port */
 
-				if (mask & BIT(SCS_CLASS_TYPE4_DSCP)) { /* DSCP */
+				if (mask & BIT(CLASS_TYPE4_DSCP)) { /* DSCP */
 					if (version == IPV4)
-						scs_add_req->type4Params.u.v4.dscp = *pos++;
+						scs_add_req->type4Params.u.v4.dscp = *pos;
 					else
-						scs_add_req->type4Params.u.v6.dscp = *pos++;
+						scs_add_req->type4Params.u.v6.dscp = *pos;
 				}
+				pos ++;
 
-				if (mask & BIT(SCS_CLASS_TYPE4_PROTO)) { /* Protocol or next header */
+				if (mask & BIT(CLASS_TYPE4_PROTO)) { /* Protocol or next header */
 					if (version == IPV4)
-						scs_add_req->type4Params.u.v4.proto = *pos++;
+						scs_add_req->type4Params.u.v4.proto = *pos;
 					else
-						scs_add_req->type4Params.u.v6.nextHeader = *pos++;
+						scs_add_req->type4Params.u.v6.nextHeader = *pos;
 				}
+				pos ++; /* Length of protocol */
+				if (version == IPV4) /* Length of reserve field in ipV4 */
+				  pos ++;
 
-				if (mask & BIT(SCS_CLASS_TYPE4_FLOW_LABEL)) { /* Flow label only for ipv6 */
+				if (mask & BIT(CLASS_TYPE4_FLOW_LABEL)) { /* Flow label only for ipv6 */
 						memcpy(scs_add_req->type4Params.u.v6.flowLabel, pos, 3);
-						pos += 3;
 				}
+				pos += 3; /* Length of flow label for ipv6 */
 
-				has_type4 = true;
-				scs_add_req->tclasInfo.tclasElemstype = 4;
-				elems_num ++;
+				scs_add_req->tclasInfo.tclasElemstype = CLASSIFIER_TYPE4;
+				(*elems_num)++;
 				break;
-			 case SCS_CLASSIFIER_TYPE10:
+			 case CLASSIFIER_TYPE10:
 				scs_add_req->type10Params.protoInstance = *pos++;
 				scs_add_req->type10Params.protoNum = *pos++;
 				scs_add_req->type10Params.filterLen = (tclas_len - 4) / 2;
@@ -384,7 +398,7 @@ int mxl_scs_parse_tclas_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req
 				memcpy(scs_add_req->type10Params.filterMask, pos, scs_add_req->type10Params.filterLen);
 				pos += scs_add_req->type10Params.filterLen;
 				has_type10 = true;
-				elems_num ++;
+				(*elems_num)++;
 				break;
 			 default:
 				return WLAN_STATUS_UNSPECIFIED_FAILURE;
@@ -395,35 +409,29 @@ int mxl_scs_parse_tclas_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req
 	 * appears before class 4. So, has_type10 and has_type6 variables are used *
 	 * and the tclas type 10 without tclas 4 is checked after the loop ends */
 	if (has_type10) {
-		if (!has_type4)
+		if (scs_add_req->tclasInfo.tclasElemstype != CLASSIFIER_TYPE4)
 			return WLAN_STATUS_UNSPECIFIED_FAILURE;
+
 		/* TCLAS type 10 with a protocol different than 17 or 50 is illegal */
 		if (scs_add_req->type10Params.protoInstance != 0 ||
 		    (scs_add_req->type10Params.protoNum != 17 &&
 		    scs_add_req->type10Params.protoNum != 50))
 			return WLAN_STATUS_UNSPECIFIED_FAILURE;
-		if (version == IPV4) {
-			if (scs_add_req->type4Params.u.v4.srcPort != 0)
-				return WLAN_STATUS_UNSPECIFIED_FAILURE;
-		}
-		else if (version == IPV6) {
-			if (scs_add_req->type4Params.u.v6.srcPort != 0)
-				return WLAN_STATUS_UNSPECIFIED_FAILURE;
-		}
+
 		scs_add_req->tclasInfo.tclasElemstype = 10;
 	}
+
 	return WLAN_STATUS_SUCCESS;
 }
 
-int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len)
+int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, const u8 *pos, size_t len, u8 scsId)
 {
-	const u8 *pos, *end;
+	const u8 *end;
 	u8 eid, elen;
 	struct sta_info *sta;
 	struct mxl_scs_add_req scs_add_req;
-	pos = mgmt->u.action.u.scs_action_req.variable;
 	u8 tclas_elems_num = 0;
-	bool has_tclas_elems = false, has_tclas_process = false;
+	bool qos_characteristic_elems = false;
 
 	os_memset(&scs_add_req, 0, sizeof(struct mxl_scs_add_req));
 
@@ -438,19 +446,18 @@ int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt
 		return WLAN_STATUS_UNSPECIFIED_FAILURE;
 	}
 
-	/* WLANRTSYS-91362: Workaround, Extra 8 bytes are appened to the frame from the lower layers */
-	len = len - 8;
-	end = (const u8 *) mgmt + len;
+	end = pos + len;
 	scs_add_req.aid = sta->aid;
 	scs_add_req.status = 0;
-	scs_add_req.scsid = mgmt->u.action.u.scs_action_req.scsId;
+	scs_add_req.scsid = scsId;
 
 	while (pos + 2 <= end) {
 		eid = *pos++;
 		elen = *pos++;
 
-		if (end - pos < elen)
+		if (end - pos < elen) {
 			return WLAN_STATUS_UNSPECIFIED_FAILURE;
+		}
 
 		switch (eid) {
 		case WLAN_EID_INTRA_ACCESS_CATEGORY_PRIORITY:
@@ -459,18 +466,17 @@ int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt
 		case WLAN_EID_TCLAS:
 			if (mxl_scs_parse_tclas_elems(pos, &scs_add_req, elen, &tclas_elems_num))
 				return WLAN_STATUS_UNSPECIFIED_FAILURE;
-			has_tclas_elems = true;
 			break;
 		case WLAN_EID_TCLAS_PROCESSING:
 			/* the processing field = 0 , indicates it should match to *
 			 * all the parameters associated with TCLAS elems */
 			if (*pos != 0 || tclas_elems_num <=1)
 			   return WLAN_STATUS_UNSPECIFIED_FAILURE;
-			has_tclas_process = true;
 			break;
 		case WLAN_EID_EXTENSION:
 			if (mxl_scs_parse_qos_characteristics_elems(pos, &scs_add_req))
 				return WLAN_STATUS_UNSPECIFIED_FAILURE;
+			qos_characteristic_elems = true;
 			break;
 		case WLAN_EID_VENDOR_SPECIFIC:
 			/*Need to verify, EID populated as optional subelement *
@@ -485,9 +491,12 @@ int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt
 		pos += elen;
 	}
 
-	/* TCLAS elems without TCLAS_PROCESSING elems is illegal */
-	if (has_tclas_elems && !has_tclas_process)
-		return WLAN_STATUS_UNSPECIFIED_FAILURE;
+	/* The direction bit is part of qos characteristics IE. *
+	 * For the DL QM SCS TCs 4.5.1 and 4.5.2, QoS IE not present. *
+	 * For the UL EHT TC 4.47.1, populates the QoS IE. *
+	 * As of now to pass the TCs, assumption is that if no QoS IE, then direction is DL */
+	if (!qos_characteristic_elems)
+		scs_add_req.direction = QOS_CTRL_INFO_DL;
 
 	if (MXL_DRV_VENDOR_SET(LTQ_NL80211_VENDOR_SUBCMD_SCS_ADD_REQ,
 				&scs_add_req, sizeof(struct mxl_scs_add_req))) {
@@ -500,7 +509,7 @@ int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt
 }
 
 int mxl_scs_remove_req(struct hostapd_data *hapd,
-			const struct ieee80211_mgmt *mgmt, size_t len)
+			const struct ieee80211_mgmt *mgmt, size_t len, u8 scsId)
 {
 	struct sta_info *sta;
 	struct mxl_scs_rem_req scs_rem_req;
@@ -517,7 +526,7 @@ int mxl_scs_remove_req(struct hostapd_data *hapd,
 	}
 
 	scs_rem_req.aid = sta->aid;
-	scs_rem_req.scsid = mgmt->u.action.u.scs_action_req.scsId;
+	scs_rem_req.scsid = scsId;
 	if (MXL_DRV_VENDOR_SET(LTQ_NL80211_VENDOR_SUBCMD_SCS_REM_REQ,
 	    &scs_rem_req, sizeof(struct mxl_scs_rem_req))) {
 		wpa_printf(MSG_ERROR, "send LTQ_NL80211_VENDOR_SUBCMD_SEND_SCS_REM_REQ failed!!!");
@@ -533,37 +542,59 @@ void mxl_handle_scs_req(struct hostapd_data *hapd,
 			const struct ieee80211_mgmt *mgmt, size_t len)
 {
 	int status;
-	u8 eid = mgmt->u.action.u.scs_action_req.eid;
-	if(eid != WLAN_EID_SCS_DESCRIPTOR) {
-		wpa_printf(MSG_ERROR, "SCS: mxl_handle_scs_req - SCS Descriptor element is not present");
-		return;
-	}
-	u8 reqType =mgmt->u.action.u.scs_action_req.reqType;
-	wpa_printf(MSG_DEBUG, "SCS req_type =%d",reqType);
-	switch(reqType)
-	{
-		case SCS_REQ_ADD:
-			status = mxl_scs_add_req(hapd, mgmt,len);
-			mxl_send_scs_resp(hapd, mgmt, status);
-			break;
-		case SCS_REQ_REMOVE:
-			status = mxl_scs_remove_req(hapd, mgmt, len);
-			mxl_send_scs_resp(hapd, mgmt, status);
-			break;
-		case SCS_REQ_CHANGE:
-			status = mxl_scs_remove_req(hapd, mgmt, len);
-			if (!status)
-				mxl_send_scs_resp(hapd, mgmt, status);
-			status = mxl_scs_add_req(hapd, mgmt,len);
-			mxl_send_scs_resp(hapd, mgmt, status);
-			break;
-		default:
-			hostapd_logger(hapd, mgmt->sa, HOSTAPD_MODULE_IEEE80211,
+	const u8 *end;
+	u8 reqType, scsId, eid;
+	size_t elen;
+	const u8 *pos = mgmt->u.action.u.scs_action_req.variable;
+	struct mxl_scs_status_list status_list[8]; /* Assuming max 8 status duples for SCS response */
+	u8 scs_count = 0;
+
+	end = ((const u8 *) mgmt) + len;
+
+	while (pos + 2 <= end) {
+		eid = *pos++;
+		if(eid != WLAN_EID_SCS_DESCRIPTOR) {
+			wpa_printf(MSG_ERROR, "SCS: mxl_handle_scs_req - SCS Descriptor element is not present");
+			return;
+		}
+		elen = *pos++;
+		scsId = *pos ++;
+		elen--; /* Subtract scsId len from elen */
+		reqType = *pos ++;
+		elen--; /* Subtract reqType len from elen */
+		switch(reqType)
+		{
+			case SCS_REQ_ADD:
+				status = mxl_scs_add_req(hapd, mgmt, pos, elen, scsId);
+				status_list[scs_count].scsId = scsId;
+				status_list[scs_count].status = status;
+				scs_count ++;
+				break;
+			case SCS_REQ_REMOVE:
+				status = mxl_scs_remove_req(hapd, mgmt, elen, scsId);
+				status_list[scs_count].scsId = scsId;
+				status_list[scs_count].status = status;
+				scs_count ++;
+				break;
+			case SCS_REQ_CHANGE:
+				status = mxl_scs_remove_req(hapd, mgmt, elen, scsId);
+				if (status == WLAN_STATUS_TCLAS_PROCESSING_TERMINATED)
+					status = mxl_scs_add_req(hapd, mgmt, pos, elen, scsId);
+				status_list[scs_count].scsId = scsId;
+				status_list[scs_count].status = status;
+				scs_count ++;
+				break;
+			default:
+				hostapd_logger(hapd, mgmt->sa, HOSTAPD_MODULE_IEEE80211,
 			        HOSTAPD_LEVEL_DEBUG,
 			        "hostapd_scs_req - unknown request type %d",
 			        reqType);
 			return;
+		}
+
+		pos += elen;
 	}
+	mxl_send_scs_resp(hapd, mgmt, status_list, scs_count);
 }
 
 void mxl_hostapd_scs_action(struct hostapd_data *hapd,
@@ -573,7 +604,7 @@ void mxl_hostapd_scs_action(struct hostapd_data *hapd,
 
 	if (!hapd->conf->mxl_bss_conf.scs_enable) {
       wpa_printf(MSG_ERROR, "SCS: mxl_scs_add_req - not enabled/supported");
-      return WLAN_STATUS_UNSPECIFIED_FAILURE;
+      return;
    }
 
 	switch (*action) {
diff --git a/src/ap/mxl_wmm.h b/src/ap/mxl_wmm.h
index 49b1225ac..359465af7 100644
--- a/src/ap/mxl_wmm.h
+++ b/src/ap/mxl_wmm.h
@@ -27,8 +27,8 @@ void mxl_hostapd_eht_protected(struct hostapd_data *hapd,
 #ifdef CONFIG_SCS_MXL
 void mxl_hostapd_scs_action(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len);
 void mxl_handle_scs_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len);
-int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len);
-int mxl_scs_remove_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len);
+int mxl_scs_add_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, const u8 *pos, size_t len, u8 scsId);
+int mxl_scs_remove_req(struct hostapd_data *hapd, const struct ieee80211_mgmt *mgmt, size_t len, u8 scsId);
 int mxl_scs_parse_qos_characteristics_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req);
 int mxl_scs_parse_tclas_elems(const u8 *pos, struct mxl_scs_add_req *scs_add_req, size_t len, u8 *elems_num);
 #endif /* CONFIG_SCS_MXL */
diff --git a/src/common/ieee802_11_defs.h b/src/common/ieee802_11_defs.h
index d1c292dca..75947477d 100644
--- a/src/common/ieee802_11_defs.h
+++ b/src/common/ieee802_11_defs.h
@@ -1185,18 +1185,13 @@ struct ieee80211_mgmt {
 				struct {
 					u8 action;
 					u8 dialog_token;
-					u8 eid;
-					u8 length;
-					u8 scsId;
-					u8 reqType;
 					u8 variable[];
 				}STRUCT_PACKED scs_action_req;
 				struct {
 					u8 action;
 					u8 dialog_token;
 					u8 count;
-					u8 scsId;
-					u16 status;
+					u8 scs_status[];
 				}STRUCT_PACKED scs_action_resp;
 #endif /* CONFIG_SCS_MXL */
 			} u;
diff --git a/src/common/mxl_ieee802_11_defs.h b/src/common/mxl_ieee802_11_defs.h
index a87408867..0233fc974 100644
--- a/src/common/mxl_ieee802_11_defs.h
+++ b/src/common/mxl_ieee802_11_defs.h
@@ -109,8 +109,8 @@ struct rnr_neighbor_ap_info {
 #define SCS_INTRA_ACCESS_USER_PRIO_MASK 0x7
 #define SCS_INTRA_ACCESS_ALTERNATE_QUEUE BIT(3)
 #define SCS_INTRA_ACCESS_DROP_ELIGIBILITY BIT(4)
-#define SCS_CLASSIFIER_TYPE4	4
-#define SCS_CLASSIFIER_TYPE10	10
+#define CLASSIFIER_TYPE4	4
+#define CLASSIFIER_TYPE10	10
 #define IPV4	4
 #define IPV6	6
 
@@ -143,15 +143,15 @@ enum wfa_capabilities_element{
 	WFA_QOS_MGMT_SCS_TRAFFIC_DESC = BIT(2),
 };
 
-enum scs_classifier_type4_mask {
-	SCS_CLASS_TYPE4_VERSION = 0,
-	SCS_CLASS_TYPE4_SRC_IP = 1,
-	SCS_CLASS_TYPE4_DST_IP = 2,
-	SCS_CLASS_TYPE4_SRC_PORT = 3,
-	SCS_CLASS_TYPE4_DST_PORT = 4,
-	SCS_CLASS_TYPE4_DSCP = 5,
-	SCS_CLASS_TYPE4_PROTO = 6,
-	SCS_CLASS_TYPE4_FLOW_LABEL = 7,
+enum classifier_type4_mask {
+	CLASS_TYPE4_VERSION = 0,
+	CLASS_TYPE4_SRC_IP = 1,
+	CLASS_TYPE4_DST_IP = 2,
+	CLASS_TYPE4_SRC_PORT = 3,
+	CLASS_TYPE4_DST_PORT = 4,
+	CLASS_TYPE4_DSCP = 5,
+	CLASS_TYPE4_PROTO = 6,
+	CLASS_TYPE4_FLOW_LABEL = 7,
 };
 
 #endif /* CONFIG_SCS_MXL */
-- 
2.43.0

